torch/_refs/__init__.py

```diff
@@ -3649,6 +3666,14 @@ def _reshape_view_helper(a: TensorLikeType, *shape, allow_copy: bool) -> TensorL
         else:
             return _a
 
+    if a.is_contiguous():
+        # Special-cases for nd_to_1d
+        if len(shape) == 1:
+            return _nd_to_1d(a)
+        # Special-cases for 1d_to_2d
+        if len(shape) == 2 and guard_size_oblivious(a.ndim == 1):
```

Comment:
There's no point in guard_size_oblivious on ndim, dimensionality of tensors cannot be dynamic

Reply:
Good point. Will fix.